wgsl/index.bs

```diff
@@ -8556,19 +8552,28 @@ path: syntax/align_attr.syntax.bs.include
 
     [=shader-creation error|Must=] only be applied to a member of a [=structure=] type.
 
-    Note: This attribute influences how a value of the enclosing structure type can appear in memory:
-    at which byte addresses the structure itself and its component members can appear.
-    In particular, the rules in [[#memory-layouts]] combine to imply the following constraint:
+    This attribute influences how a value of the enclosing structure type can appear in memory:
+    it constrains the byte addresses at which the structure itself and its component members can appear.
 
-    <p class="note" algorithm="implied constraint on align attribute">
+    <div algorithm="constraint on align attribute">
     If `align(`|n|`)` is applied to a member of |S|
-    with type |T|, and |S| is the [=store type=]
-    or contained in the store type for a variable in address space |C|,
+    with type |T|,
+    and |S| can be the [=store type=] for variable in address space |AS|,
+    where |AS| is not [=address spaces/uniform=],
     then |n| [=shader-creation error|must=] satisfy:
-    |n|&nbsp;=&nbsp;|k|&nbsp;&times;&nbsp;[=RequiredAlignOf=](|T|,|C|)
+    <blockquote>
+    |n|&nbsp;=&nbsp;|k|&nbsp;&times;&nbsp;[=RequiredAlignOf=](|T|,|AS|)
```

Comment:
Could we not just use `AlignOf()` here instead of `RequiredAlignOf()`, to make this rule address-space agnostic? Then you wouldn't need the `!uniform` condition.

Reply:
ah yes, that works. It works because it recurses to a nested type.

Reply:
Fixed this now.

Reply:
I don't think so, because we should plan for a future when we do scalar block layout.
We need to keep Align(..) only about setting layouts, and use RequiredAlignOf to be the constraints the environment places on memory accesses.